Orange/widgets/unsupervised/tests/test_owmanifoldlearning.py

```diff
@@ -73,3 +73,25 @@ def test_sparse_data(self):
         self.send_signal("Data", None)
         self.widget.apply_button.button.click()
         self.assertFalse(self.widget.Error.sparse_not_supported.is_shown())
+
+    def test_data_which_makes_singular(self):
+        """
+        Sometimes singular matrix is calculated from input data and
+        that should be handled.
+        GH-2228
+        """
+        table = Table(
+            Domain(
+                [ContinuousVariable("a"), ContinuousVariable("b")],
+                class_vars=DiscreteVariable("c", values=["0", "1"])),
+            list(zip(
+                [1, 1, 1],
+                [0, 1, 2],
+                [0, 1, 1]))
```

Comment:
Yes, matrix with all zeros would work here as well. But I just wanted to point out that regular matrix can crash the code as well.